### PR TITLE
[27] Implement post job page

### PR DESCRIPTION
**Pull request by pisuttun**
## What did you do

- Add Post job page

- Move detail job page to `modules/job/detail`

## Demo
https://dev.modela.miello.dev/job/post (log in as casting)
image

## Remark
- doesn't connect to API yet

- will write unit test later since there is a task blocked by this

- cant figure out how to change icon color in DatePicker ;w;

### CODE REVIEW

File: apps/web/src/modules/job/post/hooks/useJobForm/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    control,
+    name: 'shooting',
+  })
+
+  const handleSuccess: SubmitHandler<IPostJobSchemaType> = useCallback(() => {
+    console.log('success')
+  }, [])
+
+  const handleClickSubmit: FormEventHandler<HTMLFormElement> =
+    handleSubmit(handleSuccess)
```
Comment (miello) [inline apps/web/src/modules/job/post/hooks/useJobForm/index.ts:37]: useMemo might help some optimization here

File: apps/web/src/modules/job/post/components/ShootingForm/types.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export interface ShootingFormProps {
+  id: string
+  index: number
+  remove: (idx: number) => void
+  //pls forgive me for this
+  control: any
```
Comment (miello) [inline apps/web/src/modules/job/post/components/ShootingForm/types.ts:6]: I think we can specify this type (but not in easy way)

File: apps/web/src/modules/job/post/hooks/useJobForm/schema.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        code: 'custom',
+        path: ['endDate'],
+        message: 'วันที่สิ้นสุดการถ่ายทำต้องมากกว่าวันที่เริ่มต้น',
+      })
+    }
+  })
+
+const postJobSchema = z
+  .object({
+    jobName: z.string().min(1, 'กรุณากรอกตำแหน่งงาน'),
```
Comment (miello) [inline apps/web/src/modules/job/post/hooks/useJobForm/schema.ts:35]: Required criteria will trigger on `.string` first so we need to specify message here
```suggestion
    jobName: z.string({ required_error: 'กรุณากรอกตำแหน่งงาน' }),
```

Comment (miello) [reply apps/web/src/modules/job/post/hooks/useJobForm/schema.ts:35]: Please apply to all relevant

**Review by miello — CHANGES_REQUESTED** [review]

File: apps/web/src/common/components/TextField/index.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import { TextField as TextFieldMUI, TextFieldProps } from '@mui/material'
```
Comment (PKhing) [inline apps/web/src/common/components/TextField/index.tsx:1]: wait @miello to merge first then delete this file?

File: apps/web/src/modules/job/post/components/ShootingForm/types.ts (hunk 2 above)
Comment (PKhing) [reply apps/web/src/modules/job/post/components/ShootingForm/types.ts:6]: `Control<IPostJobSchemaType>`

**Review by PKhing — CHANGES_REQUESTED** [review]

**Comment by PKhing** [comment]
image
application deadline should be after today

**Comment by PKhing** [comment]
and somehow the form is too wide ? compare to @miello 's register form

**Comment by PKhing** [comment]
If I start working on edit job page, I will be moving a lot of files, so that you won't be able to work on it.
Do you want to connect to API first or want me to implement edit job page first?

**Review by PKhing — CHANGES_REQUESTED** [review]
-Application deadline can still be before today (in the form validation)
- All numeric field accept negative number :joy:
image
- Number of actor can be float
image
- Somehow there is no error text displayed when shooting end date is before shooting start date
image

**Review by PKhing — APPROVED** [review]

File: apps/web/src/modules/job/post/hooks/useJobForm/schema.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  })
+
+const postJobSchema = z
+  .object({
+    jobName: z.string({ required_error: 'กรุณากรอกตำแหน่งงาน' }),
+    jobDescription: z.string({ required_error: 'กรุณากรอกรายละเอียดงาน' }),
+    dueDate: z
+      .instanceof(dayjs as unknown as typeof Dayjs)
+      .refine((arg) => arg.isAfter(dayjs()), {
+        message: 'วันที่สิ้นสุดการรับสมัครต้องมากกว่าวันที่ปัจจุบัน',
```
Comment (miello) [inline apps/web/src/modules/job/post/hooks/useJobForm/schema.ts:43]:
```suggestion
        message: 'วันที่สิ้นสุดการรับสมัครต้องอยู่หลังวันที่ปัจจุบัน',
```

File: apps/web/src/modules/job/post/hooks/useJobForm/schema.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import dayjs, { type Dayjs } from 'dayjs'
+import { z } from 'zod'
+
+const shootingSchema = z
+  .object({
+    location: z.string({ required_error: 'กรุณากรอกสถานที่ถ่ายทำ' }),
+    startDate: z
+      .instanceof(dayjs as unknown as typeof Dayjs)
+      .refine((arg) => arg.isAfter(dayjs()), {
+        message: 'วันที่เริ่มต้นการถ่ายทำต้องมากกว่าวันที่ปัจจุบัน',
```
Comment (miello) [inline apps/web/src/modules/job/post/hooks/useJobForm/schema.ts:11]:
```suggestion
        message: 'วันที่เริ่มต้นการถ่ายทำต้องอยู่หลังวันที่ปัจจุบัน',
```

**Review by miello — CHANGES_REQUESTED** [review]

**Pull request merged**